crates/node/src/init.rs

```diff
@@ -164,6 +168,10 @@ impl<'a> NodeInit<'a> {
                     match err {
                         JoinError::MissingNodesConfiguration => true,
                         JoinError::ConcurrentNodeRegistration(_) => false,
+                        JoinError::MetadataStore(ReadWriteError::RemoteError(RemoteError::Unimplemented)) => {
+                            error!("Received an invalid response from the metadata store. This could indicate a misconfigured client or a missing metadata-store role on one of the stores.");
+                            false
+                        },
```

Comment:
So we are failing quickly which means that we won't try any of the other addresses that might be configured, right?

Reply:
my thinking is that the `unimplemented` is an obvious sign that your configuration is wrong, and hence it's better to correct the configuration to move on. Unimplemented error is not recoverable and it's a permanent error, correct?